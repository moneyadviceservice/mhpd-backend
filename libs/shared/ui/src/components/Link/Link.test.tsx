import React from 'react';
import { render, screen } from '@testing-library/react';
import { Link } from '.';

describe('Link component', () => {
  it('renders link correctly ', () => {
    render(
      <Link data-testid="link" href="www.test.com">
        This is a link
      </Link>,
    );
    const link = screen.getByTestId('link');
    expect(link).toMatchSnapshot();
  });

  it('renders link correctly with white text ', () => {
    render(
      <Link data-testid="link" href="www.test.com" variant="whiteText">
        This is a link
      </Link>,
    );
    const link = screen.getByTestId('link');
    expect(link).toMatchSnapshot();
  });

  it('renders link correctly as button', () => {
    render(
      <Link data-testid="link" href="www.test.com" asButtonVariant="primary">
        This is a link
      </Link>,
    );
    const link = screen.getByTestId('link');
    expect(link).toMatchSnapshot();
  });

  it('renders link correctly with inline text', () => {
    render(
      <Link data-testid="link" href="www.test.com" asInlineText={true}>
        This is a link
      </Link>,
    );
    const link = screen.getByTestId('link');
    expect(link).toMatchSnapshot();
  });
});
