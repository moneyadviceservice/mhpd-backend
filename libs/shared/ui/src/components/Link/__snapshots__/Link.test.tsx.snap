// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Link component renders link correctly  1`] = `
<a
  class="inline-flex items-center gap-2 text-pink-600 visited:text-purple-700 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline hover:no-underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none"
  data-testid="link"
  href="www.test.com"
>
  This is a link
</a>
`;

exports[`Link component renders link correctly as button 1`] = `
<a
  class="border-0 text-white bg-pink-600 hover:bg-pink-800 hover:shadow-none active:bg-pink-900 active:outline-yellow-200 active:text-white t-button px-3 py-2 font-semibold rounded outline-none shadow-bottom-gray inline-flex items-center justify-between gap-2 cursor-pointer disabled:cursor-not-allowed disabled:border-slate-400 disabled:border disabled:bg-transparent disabled:text-gray-500 disabled:shadow-none focus:bg-yellow-200 focus:outline-purple-700 focus:outline-[3px] focus:outline-offset-0 focus:text-gray-800 focus:shadow-none active:shadow-none active:outline-[3px] active:outline-offset-0"
  data-testid="link"
  href="www.test.com"
>
  This is a link
</a>
`;

exports[`Link component renders link correctly with inline text 1`] = `
<a
  class="inline items-center gap-2 text-pink-600 visited:text-purple-700 focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 underline hover:no-underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none"
  data-testid="link"
  href="www.test.com"
>
  This is a link
</a>
`;

exports[`Link component renders link correctly with white text  1`] = `
<a
  class="inline-flex items-center gap-2 text-white no-underline hover:text-pink-400 visited:text-white focus:text-gray-800 focus:bg-yellow-200 focus:shadow-link-focus focus-within:outline-0 hover:underline focus:no-underline cursor-pointer active:text-gray-800 active:underline active:bg-transparent active:shadow-none"
  data-testid="link"
  href="www.test.com"
>
  This is a link
</a>
`;
